apps/passport-client/components/screens/HaloScreen/AddHaloScreen.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  HaLoNoncePCD,
+  HaLoNoncePCDArgs,
+  HaLoNoncePCDPackage,
+} from "@pcd/halo-nonce-pcd";
+import { ArgumentTypeName } from "@pcd/pcd-types";
+import { ReactNode, useCallback, useContext, useEffect, useState } from "react";
+import styled from "styled-components";
+import { DispatchContext } from "../../../src/dispatch";
+import {
+  useHasUploaded,
+  useIsDownloaded,
+} from "../../../src/useSyncE2EEStorage";
+import { err } from "../../../src/util";
+import { Button, Spacer } from "../../core";
+import { MaybeModal } from "../../modals/Modal";
+import { AppContainer } from "../../shared/AppContainer";
+import { AppHeader } from "../../shared/AppHeader";
+import { PCDCard } from "../../shared/PCDCard";
+import { SyncingPCDs } from "../../shared/SyncingPCDs";
+
+export function AddHaloScreen({
+  pk2,
+  rnd,
+  rndsig,
+}: {
+  pk2: string;
+  rnd: string;
+  rndsig: string;
+}) {
+  const [_state, dispatch] = useContext(DispatchContext);
+  const [added, setAdded] = useState(false);
+  const [pcd, setPCD] = useState<HaLoNoncePCD | undefined>(undefined);
+  const hasUploaded = useHasUploaded();
+  const isDownloaded = useIsDownloaded();
+
+  useEffect(() => {
+    const generatePCD = async () => {
+      const args: HaLoNoncePCDArgs = {
+        requestNewPCD: {
+          argumentType: ArgumentTypeName.Boolean,
+          value: false,
+        },
+        pk2: {
+          argumentType: ArgumentTypeName.String,
+          value: pk2,
+        },
+        rnd: {
+          argumentType: ArgumentTypeName.String,
+          value: rnd,
+        },
+        rndsig: {
+          argumentType: ArgumentTypeName.String,
+          value: rndsig,
+        },
+      };
```

Comment:
this is surface area for vulnerabilities - make sure these values (taken directly from query string) can not be malicious